common/pinot/pinotQueryValidator.go

```diff
@@ -242,6 +242,15 @@ func (qv *VisibilityQueryValidator) processSystemKey(expr sqlparser.Expr) (strin
 	}
 	colNameStr := colName.Name.String()
 
+	if comparisonExpr.Operator == sqlparser.LikeStr {
+		colVal, ok := comparisonExpr.Right.(*sqlparser.SQLVal)
+		if !ok {
+			return "", fmt.Errorf("right comparison is invalid: %v", comparisonExpr.Right)
+		}
+		colValStr := string(colVal.Val)
+		return processCustomString(comparisonExpr, colNameStr, colValStr), nil
```

Comment:
ProcessCustom functions are only for the column with custom attr. In your use case about WorkflowID, those function won't be re-usable. 

Here's a doc about pinot text match. This might be helpful. https://docs.pinot.apache.org/basics/indexing/text-search-support

I've also tested this query, it is working:
SELECT * 
FROM rta.rta.cadence_visibility_docstore_staging

Where domainid = '97cc379f-0a5b-49d9-bdf2-c8cfe86eda5c'
AND TEXT_MATCH(workflowid, 'test')

You can have a try following this runbook: 
https://engwiki.uberinternal.com/display/TE0CADENCE/Pinot%3A+How+to+Directly+Query+Pinot+Tables

Reply:
thanks! There was a issue with github desktop that it failed to push my changes to use text_match(). should be updated now